### ISSUE TITLE
vi-mode inputrc creates nasty repl behavior

### ISSUE THREAD

**Issue by mathpunk** [technomancy/leiningen#1809]
I've recently lost a lot of configuration data, and I'm having a problem getting lein repl to work how it did under my old setup. My inputrc sets editing-mode to vi. In the repl, that now has the effect of entering command mode (i.e., not typing anything mode) at every repl prompt. 

That's irritating enough in and of itself, but it also means that I can't do fancy things like have tmux send text to my repl, because it's in command mode so `(+ 1 1)` gets read as `+ 1 1)`. 

I've examined #465 and #1390 but I'm afraid I don't understand what the repl does well enough to figure out how to tinker with whether it's using jline or rlwrap or what, nor how to configure such things. Is there a way to fix this, or to tell lein to ignore the inputrc?

**Comment by trptcolin** [comment]
It sounds like you just want to turn off the vi-mode-ness of your inputrc? This is a jline configuration, which I believe there's an override for, but typically people want the same readline configuration across all command line apps.

**Comment by mathpunk** [comment]
Preferably, I'd override this "keep entering command mode" behavior---then I would have the same readline configuration everywhere. 

Turns out jline is configured with a $HOME/.jline.rc. There's nothing about vi command mode in https://github.com/jline/jline2/wiki/Configuration-Properties so, as a workaround, I'm disabling my .inputrc in the repl entirely by editing that file to read
    
    jline.inputrc=$HOME/.none

where .none is a blank file. I'll ask upstream.

**Comment by mathpunk** [comment]
I did not ask upstream, because the problem was in my .inputrc itself.

A high-ranking search result for enabling vi command line editing is http://vim.wikia.com/wiki/Use_vi_shortcuts_in_terminal which tells users to put these lines into their .inputrc:

    set editing-mode vi
    set keymap vi-command

But only the first line is required. The second line is present to tell a program, "The following keybindings are for command mode." My shell ignores that second line, but jline evidently reads it and takes it to mean, "Use command mode at each line."

I've removed the second line, with no effect on my usual shell, and getting the correct behavior in the repl.
